server/vcs/provider/github/converter/pr_review_event.go

```diff
@@ -1,36 +1,62 @@
 package converter
 
 import (
+	"fmt"
 	"github.com/google/go-github/v45/github"
 	"github.com/palantir/go-githubapp/githubapp"
 	"github.com/pkg/errors"
 	"github.com/runatlantis/atlantis/server/events/models"
 	"github.com/runatlantis/atlantis/server/neptune/gateway/event"
+	"time"
 )
 
 type PullRequestReviewEvent struct {
 	RepoConverter RepoConverter
+	PullConverter PullConverter
 }
 
 func (p PullRequestReviewEvent) Convert(e *github.PullRequestReviewEvent) (event.PullRequestReview, error) {
 	installationToken := githubapp.GetInstallationIDFromEvent(e)
+	if e.GetRepo() == nil {
+		return event.PullRequestReview{}, fmt.Errorf("repo is nil")
+	}
 	repo, err := p.RepoConverter.Convert(e.GetRepo())
 	if err != nil {
 		return event.PullRequestReview{}, errors.Wrap(err, "converting repo")
 	}
+
+	if e.GetPullRequest() == nil {
+		return event.PullRequestReview{}, fmt.Errorf("pull request is nil")
+	}
+	pull, err := p.PullConverter.Convert(e.GetPullRequest())
+	if err != nil {
+		return event.PullRequestReview{}, errors.Wrap(err, "converting pull request")
+	}
+
+	if e.GetSender() == nil {
+		return event.PullRequestReview{}, errors.Wrap(err, "sender is nil")
+	}
 	user := models.User{
 		Username: e.GetSender().GetLogin(),
 	}
-	prNum := e.GetPullRequest().GetNumber()
-	action := e.GetAction()
+
+	if e.GetReview() == nil {
+		return event.PullRequestReview{}, errors.Wrap(err, "review is nil")
+	}
+	state := e.GetReview().GetState()
 	ref := e.GetReview().GetCommitID()
+	eventTimestamp := e.GetReview().GetSubmittedAt()
+	if e.GetReview().GetSubmittedAt().Equal(time.Time{}) {
```

Comment:
I'm not sure I understand this

Reply:
The legacy worker relies on GH event timestamps when determining of an event is [stale](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/events/stale_command_handler.go) in comparison to the update for a specific project. We have this logic because the introduction of gateway opened us up to the opportunity of receiving duplicate events/events out of order, so we want to always run the latest changes.

Reply:
done